rpc/utils.go

```diff
@@ -0,0 +1,455 @@
+package rpc
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"regexp"
+	"strings"
+
+	conTypes "github.com/pokt-network/pocket/consensus/types"
+	"github.com/pokt-network/pocket/shared/codec"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/utility"
+	utilTypes "github.com/pokt-network/pocket/utility/types"
+)
+
+const (
+	maxPerPage = 1000
+)
+
+var (
+	paramValueRegex *regexp.Regexp
+	errNoItems      = fmt.Errorf("no items found")
+)
+
+func init() {
+	paramValueRegex = regexp.MustCompile(`value:"(.+)"`)
+}
+
+// Broadcast to the entire validator set
+func (s *rpcServer) broadcastMessage(msgBz []byte) error {
+	utilityMsg, err := utility.PrepareTxGossipMessage(msgBz)
+	if err != nil {
+		s.logger.Error().Err(err).Msg("Failed to prepare transaction gossip message")
+		return err
+	}
+
+	if err := s.GetBus().GetP2PModule().Broadcast(utilityMsg); err != nil {
+		s.logger.Error().Err(err).Msg("Failed to broadcast utility message")
+		return err
+	}
+
+	return nil
+}
+
+// checkSortDesc takes a sort string and returns whether to sort descending or not
+func checkSortDesc(sort string) bool {
+	return !strings.EqualFold(sort, "asc")
+}
+
+// getPageIndexes calculates the indexes for the page requested reading the number of items specified
+func getPageIndexes(totalItems, page, per_page int) (startIdx, endIdx, totalPages int, err error) {
+	if totalItems == 0 {
+		err = errNoItems
+		return
+	}
+	if per_page > maxPerPage {
+		err = fmt.Errorf("per_page has a max value of %d", maxPerPage)
+		return
+	}
+	if page == 0 || per_page == 0 {
+		err = fmt.Errorf("page and per_page must both be greater than 0")
+		return
+	}
+
+	totalPages = int(math.Ceil(float64(totalItems) / float64(per_page)))
+	startIdx = (page - 1) * per_page
+	if startIdx > totalItems-1 {
+		err = fmt.Errorf("starting page too high: got %d, total pages: %d", page, totalPages)
+		return
+	}
+	endIdx = (page * per_page) - 1
+	if endIdx >= totalItems {
+		endIdx = totalItems - 1 //  Last Index
+	}
+
+	return startIdx, endIdx, totalPages, nil
+}
+
+// protocolActorToRPCProtocolActor converts the coreTypes.Actor to an RPC ProtocolActor
+func protocolActorToRPCProtocolActor(actor *coreTypes.Actor) ProtocolActor {
+	return ProtocolActor{
+		Address:         actor.Address,
+		ActorType:       protocolActorToRPCActorTypeEnum(actor.ActorType),
+		PublicKey:       actor.PublicKey,
+		Chains:          actor.Chains,
+		ServiceUrl:      actor.ServiceUrl,
+		StakedAmount:    actor.StakedAmount,
+		PausedHeight:    actor.PausedHeight,
+		UnstakingHeight: actor.UnstakingHeight,
+		OutputAddr:      actor.Output,
+	}
+}
+
+// txResultToRPCTransaction converts the txResult protobuf into the RPC Transaction type
```

Comment:
Let's be consistent and avoid `tx` and `Transaction` in the same function name. DItto below

e.g. `txResultToRPCTx`